src/api/ecom-api.ts

```diff
@@ -47,100 +41,196 @@ function getWixClient() {
     });
 }
 
-function createApi() {
+function createApi(): EcomAPI {
     const wixClient = getWixClient();
 
     return {
-        getProductsByCategory: async (categorySlug: string) => {
-            const getCategoryResult = await wixClient.collections.getCollectionBySlug(categorySlug);
-
-            return (
-                await wixClient.products
+        async getProductsByCategory(categorySlug) {
+            try {
+                const category = (await wixClient.collections.getCollectionBySlug(categorySlug)).collection;
+                if (!category) {
+                    throw new Error('Category not found');
+                }
+                let productsResponse = await wixClient.products
                     .queryProducts()
-                    .hasSome('collectionIds', [getCategoryResult.collection?._id])
-                    .find()
-            ).items;
+                    .hasSome('collectionIds', [category!._id])
+                    .limit(100)
+                    .find();
+
+                const allProducts = productsResponse.items;
+
+                // load all available products. if you have a lot of projects in your site
+                // please implement proper implementation of pagination
+                while (productsResponse.hasNext()) {
+                    productsResponse = await productsResponse.next();
+                    allProducts.push(...productsResponse.items);
+                }
+
+                return successResponse(allProducts);
+            } catch (e) {
+                return failureResponse(EcomApiErrorCodes.GetProductsFailure, getErrorMessage(e));
+            }
         },
-        getPromotedProducts: async () => {
-            return (await wixClient.products.queryProducts().limit(4).find()).items;
+
+        async getPromotedProducts() {
+            try {
+                const products = (await wixClient.products.queryProducts().limit(4).find()).items;
+                return successResponse(products);
+            } catch (e) {
+                return failureResponse(EcomApiErrorCodes.GetProductsFailure, getErrorMessage(e));
+            }
```

Comment:
We won't use this function, you can remove it.

Reply:
We won't. What about users who want to built something based on our template?